src/elections/services/elections.service.ts

```
@@ -1,52 +1,82 @@
+  private static readonly VOTER_CONTACT_MULTIPLIER = 5
+  private static readonly WIN_NUMBER_MULTIPLIER = 0.5
+  private static readonly API_VERSION = 'v1'
+
   private readonly logger = new Logger(ElectionsService.name)
 
-  private async fetchProjectedTurnout(
-    brPositionId: string,
-  ): Promise<ProjectedTurnout | null> {
-    const params = new URLSearchParams({ brPositionId })
-    const apiVersion = 'v1'
+  constructor(
+    private readonly httpService: HttpService
+  ) {
+    if (!ElectionsService.BASE_URL) {
+      throw new Error(`Please set ELECTION_API_URL in your .env. 
+        Recommendation is to point it at dev if you are developing`)
+    }
+  }
+
+  private async electionApiGet<Res, Q extends object>(
+    path: string,
+    query?: Q,
+  ): Promise<Res | null> {
+    const url = new URL(
+      `${ElectionsService.BASE_URL}/${ElectionsService.API_VERSION}/${path}`,
+    )
+
     try {
-      const response = await fetch(
-        `${ELECTION_API_URL}/${apiVersion}/projectedTurnout?${params.toString()}`,
-      )
-      return response.ok ? ((await response.json()) as ProjectedTurnout) : null
-    } catch (err) {
-      this.logger.debug(`Error: Couldn't fetch projected turnout - ${err}`)
+      const { data, status } = await lastValueFrom(this.httpService.get(
+        `${ElectionsService.BASE_URL}/${ElectionsService.API_VERSION}/${path}`,
+        {
+          params: query,
+          paramsSerializer: params =>
+            Object.entries(params)
+              .filter(([, v]) => v !== undefined && v !== null)
+              .map(([k, v]) => `${k}=${encodeURIComponent(String(v))}`)
+              .join('&'),
+        }
+      ))
+      if (status >= 200 && status < 300) return data
+        this.logger.warn(
+          `Election API GET ${path}} responded ${status}`,
+        )
+        return null
+    } catch (error) {
+      this.logger.error(`Election API GET ${path} failed: ${error}`)
       return null
     }
   }
```

Did you mean for this to fail silently apart from logging?

I the `gp-api`, project we typically throw a `502 Bad Gateway` to ensure that the error condition is exposed up the stack.

https://github.com/thegoodparty/gp-api/blob/5a8849c860b179f5c9a7f94998da7a1f7d7d5431/src/users/services/crmUsers.service.ts#L271

---

Added, good call